extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/BuildTimeEnabledProcessor.java

```diff
@@ -72,16 +76,27 @@ void ifBuildProfile(CombinedIndexBuildItem index, BuildProducer<BuildTimeConditi
 
     @BuildStep
     void unlessBuildProfile(CombinedIndexBuildItem index, BuildProducer<BuildTimeConditionBuildItem> producer) {
-        Collection<AnnotationInstance> annotationInstances = index.getIndex().getAnnotations(UNLESS_BUILD_PROFILE);
-        for (AnnotationInstance instance : annotationInstances) {
-            String profileOnInstance = instance.value().asString();
-            boolean enabled = !ConfigUtils.isProfileActive(profileOnInstance);
-            if (enabled) {
-                LOGGER.debug("Enabling " + instance.target() + " since the profile value does not match the active profile.");
-            } else {
-                LOGGER.debug("Disabling " + instance.target() + " since the profile value matches the active profile.");
+        List<AnnotationInstance> annotationInstances = getAnnotations(index.getIndex(), UNLESS_BUILD_PROFILE,
+                UNLESS_BUILD_PROFILE_CONTAINER);
+        Map<AnnotationTarget, List<AnnotationInstance>> annotationsTarget = annotationInstances.stream()
```

Comment:
I think that we should keep the original logic and use the `AND` semantics here.